packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx

```diff
@@ -0,0 +1,688 @@
+            expect(screen.queryByRole('heading', { name: getPaymentMethodName(localeContext.language)(method) })).not.toBeInTheDocument();
+        });
+
+        it('renders custom text for Klarna hosted payment methods', () => {
+            const method = {
+                ...defaultProps.method,
+                id: PaymentMethodId.Klarna,
+            };
+
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={method}
+                />,
+            );
+
+            expect(screen.queryByRole('heading', { name: getPaymentMethodName(localeContext.language)(method) })).not.toBeInTheDocument();
+        });
+
+        it('renders custom text for Quadpay hosted payment methods', () => {
+            const method = {
+                ...defaultProps.method,
+                id: PaymentMethodId.Quadpay,
+            };
+
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={method}
+                />,
+            );
+
+            expect(screen.queryByRole('heading', { name: getPaymentMethodName(localeContext.language)(method) })).not.toBeInTheDocument();
+        });
+
+        it('renders custom text for Sezzle hosted payment methods', () => {
+            const method = {
+                ...defaultProps.method,
+                id: PaymentMethodId.Sezzle,
+            };
+
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={method}
+                />,
+            );
+
+            expect(screen.queryByRole('heading', { name: getPaymentMethodName(localeContext.language)(method) })).not.toBeInTheDocument();
+        });
+
+        it('renders custom text for Zip hosted payment methods', () => {
+            const method = {
+                ...defaultProps.method,
+                id: PaymentMethodId.Zip,
+            }
+
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={method}
+                />,
+            );
+
+            expect(screen.queryByRole('heading', { name: getPaymentMethodName(localeContext.language)(method) })).not.toBeInTheDocument();
+        });
+    });
+
+    describe('renders only logo for certain hosted payment methods', () => {
+        it('renders only logo for AmazonPay hosted payment methods', () => {
+            render(
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={{
+                        ...defaultProps.method,
+                        id: PaymentMethodId.AmazonPay,
+                    }}
+                />,
+            );
+
+            const logos = screen.getByRole('img');
+            const names = screen.queryByRole('heading');
+
+            expect(logos).toBeInTheDocument();
+
+            expect(names).not.toBeInTheDocument();
```

Comment:
Same here. Can we add src or text content here?

Reply:
same here